### DISCUSSION TITLE
High LA on PUT when self-heal starts

### DISCUSSION THREAD

**Discussion by sorquan** [minio/minio#17617] [Q&A]
After migrating minio cluster VMs from SSD disks to SAS, load average jumps began to appear when autostarting self-heal on cluster nodes, what causes slowing of PUT requests.

## Expected Behavior
Cluster runs on SAS disks without Load Average jumps when self-heal (or scanner?) starts.

## Current Behavior
When minio self-heal starts on VM, load average jumps in for ~30 seconds.
VM1 LA:
![image](https://github.com/minio/minio/assets/52035133/76e1ca0a-b058-4d9e-9f7b-d4206fef8090)
VM2 LA:
![image](https://github.com/minio/minio/assets/52035133/99c06f07-4203-4b33-8913-78c80a38c10f)
Self-heal graph:
![image](https://github.com/minio/minio/assets/52035133/61597183-7800-42f7-8440-29dcffc7d37e)

## Possible Solution
Implement ability to disable automatic self-heal completely.

## Context
When self-heal starts and load average increases, PUT requests to the minio cluster take about 30s instead of 0.0XXs, this slows down the application that interacts with s3.

## Regression
No

## Your Environment
* Minio version: RELEASE.2023-06-29T05-12-28Z
* Cluster: 4 nodes on VMWare VM with SAS disks in single LVM volume with XFS
* OS: Ubuntu 20.04.3 LTS
* Kernel: 5.4.0-152-generic
* Average RPS on cluster: 25

**Comment by klauspost** [comment]
You can set the speed:

```
λ mc admin config set ALIAS scanner
KEY:
scanner  manage namespace scanning for usage calculation, lifecycle, healing and more

ARGS:
speed  (default|slowest|slow|fast|fastest)  scanner speed (default: 'default')
λ mc admin config set ALIAS scanner speed=slow
```

**Reply by sorquan** [reply]
I tried changing the scanner and heal settings to infinitely high, nothing changes the situation with LA jumps

**Reply by sorquan** [reply]
On this moment: 
scanner delay=250 max_wait=1500s cycle=60m speed=slowest
heal bitrotscan=on max_sleep=1000s max_io=1

**Reply by klauspost** [reply]
`bitrotscan` is an expensive operation and you rarely need it. Disable it if your hardware is this slow.

**Reply by sorquan** [reply]
I also experimented with this setting, turning it on / off does not play any role in this situation.

**Reply by klauspost** [reply]
@sorquan Then something else is up with your system. I can't go into detailed debugging. For that I will have to refer you to our [subscriptions](https://min.io/pricing).
